OpenRA.Game/CPos.cs

```diff
@@ -19,23 +19,22 @@ namespace OpenRA
 	public struct CPos : IScriptBindable, ILuaAdditionBinding, ILuaSubtractionBinding, ILuaEqualityBinding, ILuaTableBinding, IEquatable<CPos>
 	{
 		public readonly int X, Y;
+		public readonly int Layer;
```

Comment:
My gut reaction is worry that `CPos` is now another 4 bytes bigger and that's going to slow down other bits of code since we'll have to shuffle more bytes around for a fundamental type (particularly for maps that are now paying the cost for it despite not having extra layers).

Gut assumptions about performance are usually wrong, but until I can get some numbers one way or the other: How do you feel about `byte Layer` and possibly even `short X, Y` (hypothetically, for the moment)?